client/webserver/site/src/js/charts.js

```diff
@@ -277,11 +277,17 @@ export class DepthChart extends Chart {
     this.resize(parent.clientHeight)
   }
 
+  // setZoomBttns creates new regions for zoom in and zoom out buttons. It is
+  // used in initation of the buttons and resizing.
+  setZoomBttns () {
+    this.zoomInBttn = new Region(this.ctx, new Extents(0, 0, 0, 0))
+    this.zoomOutBttn = new Region(this.ctx, new Extents(0, 0, 0, 0))
+  }
+
   /* resized is called when the window or parent element are resized. */
   resized () {
     // The button region extents are set during drawing.
-    this.zoomInBttn = new Region(this.ctx, new Extents(0, 0, 0, 0))
-    this.zoomOutBttn = new Region(this.ctx, new Extents(0, 0, 0, 0))
+    this.setZoomBttns()
```

Comment:
I suppose this makes sense you could get here before the `DepthChart` constructor is finished because the parent class (`Chart`) `resize` calls the child's `resized`?  A little confusing. I wonder if there's a better pattern to prevent this possibility.  But this works...

Reply:
I'm sure there's a better way to design this... but I don't know it offhand.

Reply:
This seems more correct... to me. Make sure it is initiated before draw and we don't need to check later.